pkg/blockdeviceclaim/v1alpha1/build.go

```diff
@@ -253,24 +246,8 @@ func (b *Builder) WithCapacity(capacity string) *Builder {
 
 // WithOwnerReference sets the OwnerReference field in BDC with required
 //fields
-func (b *Builder) WithOwnerReference(spc *apis.StoragePoolClaim) *Builder {
-	if spc == nil {
-		b.errs = append(
-			b.errs,
-			errors.New("failed to build BDC object: spc object is nil"),
-		)
-		return b
-	}
-	trueVal := true
-	reference := metav1.OwnerReference{
-		APIVersion:         APIVersion,
-		Kind:               StoragePoolKind,
-		UID:                spc.ObjectMeta.UID,
-		Name:               spc.ObjectMeta.Name,
-		BlockOwnerDeletion: &trueVal,
-		Controller:         &trueVal,
-	}
-	b.BDC.Object.OwnerReferences = append(b.BDC.Object.OwnerReferences, reference)
+func (b *Builder) WithOwnerReference(owner *metav1.OwnerReference) *Builder {
```

Comment:
This can be passed as value, instead of passing the pointer so it can be used directly.

Reply:
Changed it.